lib/routes/qr_scan.dart

```diff
@@ -54,85 +58,136 @@ class QRScanState extends State<QRScan> {
           Positioned(
             right: 10,
             top: 5,
-            child: Container(
-              child: IconButton(
-                padding: const EdgeInsets.fromLTRB(0, 32, 24, 0),
-                icon: SvgPicture.asset(
-                  "src/icon/image.svg",
-                  color: Colors.white,
-                  width: 32,
-                  height: 32,
-                ),
-                onPressed: () async {
-                  final _picker = ImagePicker();
-                  XFile pickedFile = await _picker
-                      .pickImage(source: ImageSource.gallery)
-                      .catchError((err) {});
-                  final File file = File(pickedFile.path);
-                  try {
-                    if (file == null) {
-                      return;
-                    }
-                    String data = await QrCodeToolsPlugin.decodeFrom(file.path);
-                    Navigator.of(context).pop(data);
-                  } catch (e) {}
-                },
-              ),
-            ),
+            child: const ImagePickerButton(),
           ),
           Positioned(
             bottom: 30.0,
             right: 0,
             left: 0,
             child: defaultTargetPlatform == TargetPlatform.iOS
-                ? Center(
-                    child: Container(
-                      decoration: BoxDecoration(
-                        borderRadius: BorderRadius.all(
-                          const Radius.circular(12.0),
-                        ),
-                        border: Border.all(
-                          color: Colors.white.withOpacity(0.8),
-                        ),
-                      ),
-                      child: TextButton(
-                        style: TextButton.styleFrom(
-                          padding: const EdgeInsets.only(right: 35, left: 35),
-                        ),
-                        onPressed: () => Navigator.of(context).pop(),
-                        child: Text(
-                          texts.qr_scan_action_cancel,
-                          style: TextStyle(
-                            color: Colors.white,
-                          ),
-                        ),
-                      ),
-                    ),
-                  )
-                : SizedBox(),
-          )
+                ? const QRScanCancelButton()
+                : const SizedBox(),
+          ),
+          const ScanOverlay(),
         ],
       ),
     );
   }
+  /*
```

Comment:
I believe this commented code can be deleted right?

Reply:
Yes, they were added for demonstration purposes. We were discussing if these features would be beneficial. I will remove them and merge the changes. Thanks.